pkg/apiserver/alerts_test.go

```diff
@@ -93,8 +93,290 @@ func TestCreateAlert(t *testing.T) {
+
+	//test range
+	w = httptest.NewRecorder()
+	req, _ = http.NewRequest("GET", "/v1/alerts?range=ratata", nil)
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 500, w.Code)
+	assert.Equal(t, w.Body.String(), `{"message":"unable to convert 'ratata' to int interval: 'ratata' is not a valid CIDR: invalid ip address / range"}`)
+
+	//test since
+	w = httptest.NewRecorder()
+	req, _ = http.NewRequest("GET", "/v1/alerts?since=1h", nil)
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 200, w.Code)
+	assert.Contains(t, w.Body.String(), "Ip 91.121.79.195 performed 'crowdsecurity/ssh-bf' (6 events over ")
+	assert.Contains(t, w.Body.String(), `scope":"Ip","simulated":false,"start_ip":1534676931,"type":"ban","value":"91.121.79.195"`)
+
+	//test since
+	w = httptest.NewRecorder()
+	req, _ = http.NewRequest("GET", "/v1/alerts?since=1ns", nil)
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 200, w.Code)
+	assert.Equal(t, "null", w.Body.String())
+
+	//test since
+	w = httptest.NewRecorder()
+	req, _ = http.NewRequest("GET", "/v1/alerts?since=1zuzu", nil)
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 500, w.Code)
+	assert.Equal(t, w.Body.String(), `{"message":"while parsing duration: time: unknown unit zuzu in duration 1zuzu"}`)
+
+	//test until
+	w = httptest.NewRecorder()
+	req, _ = http.NewRequest("GET", "/v1/alerts?until=1ns", nil)
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 200, w.Code)
+	assert.Contains(t, w.Body.String(), "Ip 91.121.79.195 performed 'crowdsecurity/ssh-bf' (6 events over ")
+	assert.Contains(t, w.Body.String(), `scope":"Ip","simulated":false,"start_ip":1534676931,"type":"ban","value":"91.121.79.195"`)
+
+	//test until
+	w = httptest.NewRecorder()
+	req, _ = http.NewRequest("GET", "/v1/alerts?until=1m", nil)
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 200, w.Code)
+	assert.Equal(t, "null", w.Body.String())
+
+	//test until
+	w = httptest.NewRecorder()
+	req, _ = http.NewRequest("GET", "/v1/alerts?until=1zuzu", nil)
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 500, w.Code)
+	assert.Equal(t, w.Body.String(), `{"message":"while parsing duration: time: unknown unit zuzu in duration 1zuzu"}`)
+
+	//test simulated
```

Comment:
Maybe you need to add 2 decisions on the DB, one with simulated and the other one without.

Reply:
added a dedicated test `TestSimulatedAlert` with a specific dataset testing the `simulated` filter